src/api/routes/vehicles.ts

```diff
@@ -8,7 +8,33 @@ const router = express.Router();
 
 router.get("/", (req: Request, res: Response, next: NextFunction) => {
     try {
-        res.status(200).json(data);
+        const manufacturer: any = req.query.manufacturer;
+        const result = [];
+        if (manufacturer !== undefined) {
+            for (let i: any = 0; i < data.data.length; i++) {
+                if (data.data[i].manufacturer === manufacturer) {
+                    result.push(data.data[i]);
+                }
+            }
+            res.status(200).json({ data: result });
+        }
+        else {
+            res.status(200).json(data);
```

Comment:
Please return `400` if `query.manufacturer` does not match any given option.

Reply:
Isn't it wrong? 
Since its `/ ` route so if the user wont specify any query it should return the whole data.

Reply:
Say, the user hits `/vehicles?manufacturer=qwerty12345` endpoint, which is not a manufacturer of any of the available cars. Then it should return `400`